tools/user-scripts/src/lib/user-scripts.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpClient, HttpClientRequest } from '@effect/platform';
+import { Data, Effect, ManagedRuntime } from 'effect';
+import { NodeHttpClient } from '@effect/platform-node';
+
+export class UnexpectedStatus extends Data.TaggedError('UnExpectedStatus')<{
+  message: string;
+  cause: string;
+}> {}
+
+export class RequiresServerUrl extends Data.TaggedError('RequiresServerUrl')<{
+  message: string;
+  cause: string;
+}> {}
+
+export class UserService extends Effect.Service<UserService>()('@users/service', {
+  dependencies: [NodeHttpClient.layerUndici],
+  effect: Effect.gen(function* () {
+    const client = yield* HttpClient.HttpClient;
+
+    return {
+      deleteUser: (baseUrl: string, envId: string, userId: string, accessToken: string) =>
+        Effect.gen(function* () {
+          const baseRequest = HttpClientRequest.make('POST')(baseUrl).pipe(
+            HttpClientRequest.setHeader('Content-Type', 'application/json'),
+          );
+          const request = baseRequest.pipe(
+            HttpClientRequest.setMethod('DELETE'),
+            HttpClientRequest.appendUrl(`/environments/${envId}/users/${userId}`),
+            HttpClientRequest.bearerToken(accessToken),
```

Comment:
at the moment we need to pass in an access token. We could investigate alternatives. if we want to call this in our test code though, we need to have the token